moby/moby
# graphdriver/btrfs: depend on kernel UAPI instead of libbtrfs

**- What I did**
Use the kernel UAPI to provide the Btrfs headers, instead of btrfs-progs/libbtrfs

**- How I did it**
Given #44707 this was a very straightforward substitution with most of the work in research/verification.

**- How to verify it**
Verify the assumptions (e.g. kernel versions) and use containers to test building on platforms of interest (e.g. EL 7 with newer kernel headers).

**- Description for the changelog**
* The Btrfs graphdriver now requires headers from kernel 4.7 or newer. This only affects users building from source. Users on older kernels may opt to not compile this graphdriver, or to provide headers from a newer kernel.

**- A picture of a cute animal (not mandatory but encouraged)**
![image](https://user-images.githubusercontent.com/1049222/211060988-a5ab4d24-0dd7-4dc5-a2aa-cf23d71907d1.png)
